src/reader/mod.rs

```diff
@@ -821,20 +840,20 @@ trait XmlSource<'r, B> {
 
     /// Read input until processing instruction is finished.
     ///
-    /// This method expect that `<?` already was read.
+    /// This method expect that start sequence of a parser already was read.
     ///
-    /// Returns a slice of data read up to end of processing instruction (`>`),
-    /// which does not include into result (`?` at the end included).
+    /// Returns a slice of data read up to end of a chunk,  which does not include
+    /// into result.
     ///
-    /// If input (`Self`) is exhausted and nothing was read, returns `None`.
+    /// If input (`Self`) is exhausted and nothing was read, returns `SyntaxError`.
```

Comment:
"nothing was read" is likewise not very specific without the context that was removed.  Whichever thing this parser was  trying to read, was not read completely.